tests/Unit/ApparentHorizons/StrahlkorperGrTestHelpers.cpp

```diff
@@ -103,6 +108,125 @@ Scalar<DataType> horizon_radius(const std::array<DataType, 2>& theta_phi,
 
   return Scalar<DataType>{sqrt(radius_squared)};
 }
+
+template <typename DataType>
+Scalar<DataType> horizon_ricci_scalar(
+    const Scalar<DataType>& horizon_radius, const double& mass,
+    const double& dimensionless_spin_z) noexcept {
+  // Compute Kerr spin parameter a
+  // This is the magnitude of the dimensionless spin times the mass
+  double kerr_spin_a = mass * dimensionless_spin_z;
+
+  // Compute the Boyer-Lindquist horizon radius, r+
+  const double kerr_r_plus = mass + sqrt(square(mass) - square(kerr_spin_a));
+
+  // Get the Ricci scalar of the horizon, e.g. Eq. (119) of
+  // https://arxiv.org/abs/0706.0622
+  // The precise relation used here is derived in
+  // https://v2.overleaf.com/read/twdtxchyrtyv
+  auto ricci_scalar = make_with_value<Scalar<DataVector>>(horizon_radius, 0.0);
+  get(ricci_scalar) += 2.0 * (square(kerr_r_plus) + square(kerr_spin_a)) *
+                       (3.0 * square(get(horizon_radius)) -
+                        2.0 * square(kerr_r_plus) - 3.0 * square(kerr_spin_a));
+  get(ricci_scalar) /= pow(-1.0 * square(get(horizon_radius)) +
+                               square(kerr_spin_a) + 2.0 * square(kerr_r_plus),
+                           3);
+  return ricci_scalar;
+}
+
+template <typename DataType>
+Scalar<DataType> horizon_ricci_scalar(
+    const Scalar<DataType>& horizon_radius_spin_on_z_axis,
+    const YlmSpherepack& ylm_spin_on_z_axis, const double& mass,
+    const std::array<double, 3>& dimensionless_spin) noexcept {
+  // get the dimensionless spin magnitude and direction
+  const double spin_magnitude =
+      sqrt(square(dimensionless_spin[0]) + square(dimensionless_spin[1]) +
+           square(dimensionless_spin[2]));
+  const double spin_theta = acos(dimensionless_spin[2] / spin_magnitude);
```

Comment:
use atan2(y,x) instead of acos

Reply:
Can you please explain why?

spin_x = spin_magnitude * sin(spin_theta) * cos(spin_phi)
spin_y = spin_magnitude * sin(spin_theta) * sin(spin_phi)
spin_z = spin_magnitude * cos(spin_theta)

Isn't it more expensive to do 

const double spin_theta = atan2(sqrt(square(spin_x)+square(spin_y))/spin_magnitude, spin_z/spin_magnitude);

than to do

const double spin_theta = acos(spin_z / spin_magnitude)?

Reply:
Using atan2 is much more accurate than using acos for small angles.  (Unrelatedly, in your example you can leave off the divisions by spin_magnitude in atan2.)